lib/veri_policy.mli

```diff
@@ -0,0 +1,30 @@
+open Core_kernel.Std
+open Bap.Std
+open Bap_traces.Std
+
+(** rule = ACTION : INSN : EVENT : EVENT  *)
+
+type t
+type event = Trace.event
+type events = Value.Set.t
+type set_pair = events * events
+type action
+
+val skip : action
+val deny : action
+
+val create:
+  ?insn:string -> ?left:string -> ?right:string -> action -> t
+
+val match_events: t -> string -> events -> events -> 
+  (event option * event option) list
+
+(** type [deny_error] list all events that matches to deny rule *)
+type deny_error = set_pair
+
+type r = (set_pair, deny_error) Result.t
```

Comment:
There is no reason to confuse people with extra type name. `type r = (set_pair,set_pair) Result.t` is enough. But in general, I dislike the idea to reuse `Result.t` for a function that performs classification. `Result.t` is used to represent partial function, i.e., a function whose result is undefined for some inputs, in that case `Error` is returned. In your case, you are just partition input into two sets, so using `Either` here is much better fit. Even your documentation comment, prescribes this.